lib/plugins/aws/provider.js

```diff
@@ -1779,6 +1794,24 @@ class AwsProvider {
     return runtime || runtimeSourceValue.value || defaultRuntime;
   }
 
+  resolveRuntimeManagement(input) {
+    if (typeof input === 'string') {
+      return {
+        mode: input,
+      };
+    }
+    return input;
+  }
+
+  resolveFunctionRuntimeManagement(functionRuntimeManagement) {
+    const resolvedFunctionRuntimeManagement =
+      this.resolveRuntimeManagement(functionRuntimeManagement);
+    const resolvedProviderRuntimeManagement = this.resolveRuntimeManagement(
+      this.serverless.service.provider.runtimeManagement
+    );
+    return { ...resolvedProviderRuntimeManagement, ...resolvedFunctionRuntimeManagement };
```

Comment:
I think it'll be good to start object with `mode: "auto"` (so we always get the complete resolution

Reply:
The problem is that, if we start with auto, not all regions support this, so there's the potential for compatibility issues

Reply:
In my understanding, `auto` is a default, and it's how AWS will work if we do not provide `runtimeManagment`  information. Am I missing something?

Also, I'd expect that with `auto`, we do not generate `runtimeManagement` information into the CF template

Reply:
The second line is what I wasn't wrapping my head around, understood